homeassistant/components/broadlink/remote.py

```diff
@@ -281,19 +289,25 @@ async def async_learn_command(self, **kwargs):
         """Learn a list of commands from a remote."""
         kwargs = SERVICE_LEARN_SCHEMA(kwargs)
         commands = kwargs[ATTR_COMMAND]
+        command_type = kwargs[ATTR_COMMAND_TYPE]
         device = kwargs[ATTR_DEVICE]
         toggle = kwargs[ATTR_ALTERNATIVE]
 
         if not self._state:
             return
 
+        learn_command = (
+            self._async_learn_ir_command
+            if command_type == INFRARED
+            else self._async_learn_rf_command
```

Comment:
We should not allow RF if the device does not have this capability, but there is no elegant way to verify this in the current implementation. I think we should segregate these classes at the library. Another option is creating some kind of support_rf boolean at [get_devices()](https://github.com/mjg59/python-broadlink/blob/816773658868acd638c5a70fea6024ab05bdeb7d/broadlink/__init__.py#L18), which has the advantage of not breaking anything in other applications that may depend on the name of these classes.

Reply:
Use normal if else instead of inlined